src/main/java/org/terasology/customOreGen/PocketStructureDefinition.java

```diff
@@ -176,12 +179,12 @@ public void generateStructure(StructureCallback callback) {
                         }
 
                         // transform into local coordinates
-                        pos[0] = x + 0.5F;
-                        pos[1] = y + 0.5F;
-                        pos[2] = z + 0.5F;
-                        invMat.transformVector(pos);
+                        pos.x = x + 0.5F;
+                        pos.y = y + 0.5F;
+                        pos.z = z + 0.5F;
+                        invMat.transformDirection(pos);
```

Comment:
This should be just `transform` (or equivalently, `pos.mul(invMat)`), not `transformDirection`. The translation component is important too.